walletconnect/main.js

```diff
@@ -0,0 +1,179 @@
+import {
+  AccountId,
+  TokenAssociateTransaction,
+  TokenId,
+} from "@hashgraph/sdk";
+import {
+  DAppConnector,
+  HederaJsonRpcMethod,
+  HederaSessionEvent,
+  HederaChainId,
+} from "@hashgraph/hedera-wallet-connect";
+
+// Global state for wallet connection
+let accountId = '';
+let isConnected = false;
+
+// Network configuration
+const NETWORK_CONFIG = {
+  testnet: {
+    network: "testnet",
+    jsonRpcUrl: "https://testnet.hedera.validationcloud.io/v1/GYSdi5Rlhc-NmoBLSVJGsqVQDOL6C4lCCQbyHc3NvsM",
+    mirrorNodeUrl: "https://testnet.mirrornode.hedera.com",
+    chainId: "0x128",
+  },
+};
+
+const walletConnectProjectId = "377d75bb6f86a2ffd427d032ff6ea7d3";
+const currentNetworkConfig = NETWORK_CONFIG.testnet;
+const hederaNetwork = currentNetworkConfig.network;
+const metadata = {
+  name: "Hedera Vanilla JS",
+  description: "Simple Hedera WalletConnect Integration",
+  url: window.location.origin,
+  icons: [window.location.origin + "/logo192.png"],
+};
+
+// Initialize WalletConnect
+const dappConnector = new DAppConnector(
+  metadata,
+  hederaNetwork,
+  walletConnectProjectId,
+  Object.values(HederaJsonRpcMethod),
+  [HederaSessionEvent.ChainChanged, HederaSessionEvent.AccountsChanged],
+  [HederaChainId.Testnet]
+);
+
+// Ensure WalletConnect is initialized only once
+let walletConnectInitPromise = undefined;
+const initializeWalletConnect = async () => {
+  if (walletConnectInitPromise === undefined) {
+    walletConnectInitPromise = dappConnector.init();
+  }
+  await walletConnectInitPromise;
+};
+
+// Sync WalletConnect state
+function syncWalletconnectState() {
+  const account = dappConnector.signers[0]?.getAccountId()?.toString();
```

Comment:
Note that `signers[0]` may be undefined/ null.
I think you may need to use optional chaining syntax here too, e.g. `signers.?[0]`
 
Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#syntax